library/src/main/java/com/novoda/downloadmanager/lib/BatchStatusRepository.java

```diff
@@ -78,4 +78,31 @@ int getBatchStatus(long batchId) {
         return STATUS_UNKNOWN_ERROR;
 
     }
+
+    long getBatchSizeInBytes(long batchId) {
+        Cursor cursor = null;
+        long totalSize = 0;
+        try {
+            String[] selectionArgs = {String.valueOf(batchId)};
+            cursor = resolver.query(ALL_DOWNLOADS_CONTENT_URI,
```

Comment:
You can probably do a sum of the column in one go as http://stackoverflow.com/questions/21970286/android-how-to-sum-all-the-columns-in-a-sqlite-database-and-return-1-single-var